File: src/main/java/com/meetify/server/model/Place.java
package com.meetify.server.model;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

/**
 * Place class
 * Created by dmitry on 10/14/16.
 */

@SuppressWarnings("unused")
@Entity
public class Place extends JsonToString {

    private static Long currentId;
    @EmbeddedId
    private Id id;
    private String name;

    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "owner", nullable = false)
    private User owner;

    @ElementCollection
    private Set<Id> allowed;

    private Place() {

    }

    public Place(String name) {
        this.name = name;
        this.id = new Id(nextId());
        this.allowed = new HashSet<>();
    }

    public Place(String name, User user) {
        this(name);
        this.name = name;
        this.owner = user;
    }

    public static void setCurrentId(Long value) {
        if (currentId == null) {
            currentId = value;
        }
    }

    private static Long nextId() {
        return currentId++;
    }

    public User getOwner() {
        return owner;
    }

    public Set<Id> getAllowed() {
        return allowed;
    }

    public Id getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Place place = (Place) o;

        return id != null ? id.equals(place.id) : place.id == null;

    }

    @Override
    public int hashCode() {
        return id != null ? id.hashCode() : 0;
    }
}

File: src/main/java/com/meetify/server/model/Place.kt
package com.meetify.server.model

import java.io.Serializable
import java.util.*
import javax.persistence.*


/**
 * This class is place, that contains information place.
 * @author      Dmitry Baynak
 * @version     0.0.1
 * @since       0.0.1
 * @property    name        place name.
 * @property    description place description.
 * @property    id          Id.
 * @property    owner       owner id.
 * @property    photo       photo id.
 * @property    location    where is geographically place located.
 * @property    allowed     collection with users, which can view info about this place.
 * @constructor defined place's properties.
 */

@Table(name = "places")
@Entity
class Place(var name: String = "",
            var description: String = "",
            @Embedded val owner: Id = Id(-1), //todo: rename this column
            @Embedded val photo: Id = Id(-1), //todo: rename this column
            @EmbeddedId override var id: Id = Id(-1),
            @ElementCollection var allowed: Set<Id> = HashSet<Id>(),
            @Embedded var location: Location = Location()) : BaseEntity(id), Serializable